parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetrics.java

```diff
@@ -0,0 +1,613 @@
+      if (metrics != null) {
+        return ImmutableList.of(metrics);
+      }
+
+      return ImmutableList.of();
+    }
+
+    private FieldMetrics<ByteBuffer> metricsFromFieldMetrics(
+        int fieldId, org.apache.iceberg.types.Type.PrimitiveType icebergType, int truncateLength) {
+      FieldMetrics<?> fieldMetrics = metricsById.get(fieldId);
+      if (null == fieldMetrics) {
+        return null;
+      } else if (truncateLength <= 0) {
+        return new FieldMetrics<>(
+            fieldMetrics.id(),
+            fieldMetrics.valueCount(),
+            fieldMetrics.nullValueCount(),
+            fieldMetrics.nanValueCount());
+      } else {
+        Object lowerBound =
+            truncateLowerBound(icebergType, fieldMetrics.lowerBound(), truncateLength);
+        Object upperBound =
+            truncateUpperBound(icebergType, fieldMetrics.upperBound(), truncateLength);
+        ByteBuffer lower = Conversions.toByteBuffer(icebergType, lowerBound);
+        ByteBuffer upper = Conversions.toByteBuffer(icebergType, upperBound);
+        return new FieldMetrics<>(
+            fieldMetrics.id(),
+            fieldMetrics.valueCount(),
+            fieldMetrics.nullValueCount(),
+            fieldMetrics.nanValueCount(),
+            lower,
+            upper);
+      }
+    }
+
+    private FieldMetrics<ByteBuffer> metricsFromFooter(
+        int fieldId,
+        org.apache.iceberg.types.Type.PrimitiveType icebergType,
+        PrimitiveType primitive,
+        int truncateLength) {
+      if (primitive.getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.INT96) {
+        return null;
+      } else if (truncateLength <= 0) {
+        return counts(fieldId);
+      } else {
+        return bounds(fieldId, icebergType, primitive, truncateLength);
+      }
+    }
+
+    private FieldMetrics<ByteBuffer> counts(int fieldId) {
+      ColumnPath path = ColumnPath.get(currentPath());
+      long valueCount = 0;
+      long nullCount = 0;
+
+      for (ColumnChunkMetaData column : columns.get(path)) {
+        Statistics<?> stats = column.getStatistics();
+        if (stats == null || stats.isEmpty()) {
+          return null;
+        }
+
+        nullCount += stats.getNumNulls();
+        valueCount += column.getValueCount();
+      }
+
+      return new FieldMetrics<>(fieldId, valueCount, nullCount);
+    }
+
+    private <T> FieldMetrics<ByteBuffer> bounds(
+        int fieldId,
+        org.apache.iceberg.types.Type.PrimitiveType icebergType,
+        PrimitiveType primitive,
+        int truncateLength) {
+      if (icebergType == null) {
+        return null;
+      }
+
+      ColumnPath path = ColumnPath.get(currentPath());
+      Comparator<T> comparator = Comparators.forType(icebergType);
+      long valueCount = 0;
+      long nullCount = 0;
+      T lowerBound = null;
+      T upperBound = null;
+
+      for (ColumnChunkMetaData column : columns.get(path)) {
```

Comment:
This uses the column path so the same `columns` map can be used for fields with IDs (normal Iceberg fields) and for fields without IDs (shredded Variant fields).